spot-ml/src/main/scala/org/apache/spot/netflow/FlowWordCreator.scala

```diff
@@ -68,76 +60,77 @@ class FlowWordCreator(timeCuts: Array[Double],
     * @return String "word" summarizing a netflow connection.
     */
   def dstWordUDF = udf((hour: Int,
-                        minute: Int,
-                        second: Int,
                         srcIP: String,
                         dstIP: String,
                         srcPort: Int,
                         dstPort: Int,
-                        ipkt: Long,
-                        ibyt: Long) =>
-    flowWords(hour, minute, second, srcPort, dstPort, ipkt, ibyt).dstWord)
+                        protocol: String,
+                        ibyt: Long,
+                        ipkt: Long) =>
+    flowWords(hour, srcPort, dstPort, protocol, ibyt, ipkt).dstWord)
 
 
   /**
     * Calculate the source and destination words summarizing a netflow record.
     *
     * @param hour
-    * @param minute
-    * @param second
     * @param srcPort
     * @param dstPort
     * @param ipkt
     * @param ibyt
     * @return [[FlowWords]] containing source and destination words.
     */
-  def flowWords(hour: Int, minute: Int, second: Int, srcPort: Int, dstPort: Int, ipkt: Long, ibyt: Long): FlowWords = {
+  def flowWords(hour: Int, srcPort: Int, dstPort: Int, protocol: String, ibyt: Long, ipkt: Long): FlowWords = {
 
     Try {
-      val timeOfDay: Double = hour.toDouble + minute.toDouble / 60 + second.toDouble / 3600
+      val lnOf2 = scala.math.log(2) // natural log of 2
+      val ibytBin: Long =
+        scala.math.ceil(scala.math.log(ibyt) / lnOf2).toLong // 0 values should never ever happen
```

Comment:
I used floor instead of ceiling for the exponential/logarithmic binning. I think we need to choose one or the other to be consistent. However, I guess at the end it doesn't matter because either everything is shifted up or down.